libmachine/drivers/plugin/localbinary/plugin.go

```diff
@@ -89,19 +81,21 @@ func driverPath(driverName string, binaryPath string) string {
 	return driverName
 }
 
-func NewPlugin(driverName string, binaryPath string) (*Plugin, error) {
+func NewPlugin(driverName string, binaryPath string, logger Logger) (*Plugin, error) {
 	driverPath := driverPath(driverName, binaryPath)
 	binaryPath, err := exec.LookPath(driverPath)
 	if err != nil {
 		return nil, err
 	}
 
-	log.Debugf("Found binary path at %s", binaryPath)
+	logger.Debugf("Found binary path at %s", binaryPath)
 
 	return &Plugin{
+		logger: logger,
 		stopCh: make(chan struct{}),
 		addrCh: make(chan string, 1),
 		Executor: &Executor{
+			logger:     logger,
```

Comment:
I'm not a big fan of duplicating in each object something which is ultimately a shared resource and which will be the same for the whole process. This is just confusing when reading the code and trying to find out what creates this per-object thing, how this instance behaves differently from the others, ...

Reply:
The alternative is to use the `logrus.Info` etc directly. I can do and I think the output will be the same. Would it be better?

Reply:
I removed this logger and just used logrus everywhere. PTAL.